File: chargeNET/ChargeNet.Services/Database/Tariff.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ChargeNet.Services.Database
{
    public class Tariff : BaseEntity
    {
        [Required, StringLength(100)]
        public string Name { get; set; } = string.Empty;

        [Column(TypeName = "decimal(10,4)")]
        public decimal PricePerKWh { get; set; }

        [Column(TypeName = "decimal(10,4)")]
        public decimal? PricePerMinute { get; set; }

        [Required, StringLength(3)]
        public string Currency { get; set; } = "EUR";

        public TimeSpan? StartHour { get; set; }
        public TimeSpan? EndHour { get; set; }

        public bool IsActive { get; set; } = true;

        public DateTime? ValidFrom { get; set; }
        public DateTime? ValidTo { get; set; }

        // Navigation
        public virtual ICollection<ChargingSession> ChargingSessions { get; set; } = new List<ChargingSession>();
    }
}

File: chargeNET/UI/chargenet_shared/lib/models/tariff.dart
import '../api/json_utils.dart';

class Tariff {
  const Tariff({
    required this.id,
    required this.name,
    required this.pricePerKwh,
    required this.currency,
    this.pricePerMinute,
    this.isActive = true,
  });

  final int id;
  final String name;
  final double pricePerKwh;
  final String currency;
  final double? pricePerMinute;
  final bool isActive;

  factory Tariff.fromJson(Map<String, dynamic> json) {
    return Tariff(
      id: (json['id'] as num?)?.toInt() ?? 0,
      name: json['name'] as String? ?? '',
      pricePerKwh: (json['pricePerKWh'] as num?)?.toDouble() ?? 0,
      currency: json['currency'] as String? ?? 'EUR',
      pricePerMinute: (json['pricePerMinute'] as num?)?.toDouble(),
      isActive: json['isActive'] as bool? ?? true,
    );
  }

  static List<Tariff> listFromJson(dynamic json) =>
      parseJsonList(json, Tariff.fromJson);
}
